src/mckay/utilities/staticlibraries/StringMethods.java

```diff
@@ -198,6 +199,33 @@ public static int getIndexOfString(String given_name,
           throw new Exception("Unable to find " + given_name + ".");
      }
      
+	 
+	 /**
+	  * Returns all indexes of to_search which contain a String equaling to_search_for.
+	  * 
+	  * @param to_search_for	The string to search the array for.
+	  * @param to_search		The array to search.
+	  * @return					The indexes of to_search containing a string equaling to_search_for. Null if
+	  *							there are no matches.
+	  */
+	 public static int[] getAllIndexesOfString(String to_search_for, String[] to_search)
+	 {
+		 ArrayList<Integer> indexes = new ArrayList<>();

```

Comment:
Use List<Integer> indexes = new ArrayList<>() to allow for polymorphism if necessary. Seems fine in this case, but in general this should be avoided.